In [1]:
import requests
from bs4 import BeautifulSoup
products=[]
for i in range (1, 21):
  html1 = requests.get("https://r-komplekt.ru/catalog/kofemashiny_traditsionnye/?PAGEN_1="+str(i)).content.decode('utf-8')
  soup = BeautifulSoup(html1, 'html.parser')
  products.extend(soup.find_all("div", class_="product-line"))
print(len(products))





600


In [2]:
!pip install dash
!pip install dash-bootstrap-components
!pip install dash_core_components






In [3]:
import pandas as pd
import re
import matplotlib.pyplot as plt
import numpy as np
import dash
from dash import dcc, html
import plotly.express as px


In [4]:
links=[]
for product in products:
  links.append("https://r-komplekt.ru" + product.find("a")["href"])


In [5]:

country=[]
brand=[]
price=[]
setting=[]
connection=[]
coffee=[]
boiler_volume=[]
grinder=[]
frother=[]
title=[]
i=1
list1=['Страна', 'Производитель','Установка', 'Подключение','Тип кофе', 'Объём бойлера, л', 'Встроенная кофемолка', 'Капучинатор' ]
for current_link in links:
  print(i)
  html3 = requests.get(current_link).content.decode('utf-8')
  soup = BeautifulSoup(html3, 'html.parser')
  if soup.find( class_="changeShortDescription") is not None and soup.find( class_="price changePrice") is not None:
    title.append(soup.find( class_="changeShortDescription").get_text(strip=True))
    price.append(soup.find( class_="price changePrice").get_text(strip=True))
    tables=soup.find_all('table')
    correct_table=tables[0]
    headers = []
    items = []
    for header in correct_table.find_all('th'):
        headers.append(header.text.strip())
    for line in correct_table.find_all('tr'):
        cells = line.find_all('td')
        row = []
        for cell in cells:
          row.append(cell.text.strip())
        if row:
          items.append(row)
    df = pd.DataFrame(items)
    df=df.set_index(0)
    df=df.fillna("--")
    #df=df.drop(2, axis=1)
    for item in list1:
      if item not in df.index:
        new_values = {col: '—-' for col in df.columns}
        df.loc[item]=new_values
    country.append(df.loc['Страна', 1])
    brand.append(df.loc['Производитель', 1])
    setting.append(df.loc['Установка', 1])
    connection.append(df.loc['Подключение', 1])
    coffee.append(df.loc['Тип кофе', 1])
    boiler_volume.append(df.loc['Объём бойлера, л', 1])
    grinder.append(df.loc['Встроенная кофемолка', 1])
    frother.append(df.loc['Капучинатор', 1])

    i+=1
  else:
    pass


1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56
57
58
59
60
61
62
63
64
65
66
67
68
69
70
71
71
72
73
73
74
75
76
77
78
79
80
81
82
83
84
85
86
86
87
88
89
90
91
92
93
94
95
96
97
97
98
99
100
101
102
103
104
105
105
106
107
108
109
110
111
112
113
114
115
116
117
118
119
120
121
122
123
124
125
126
127
128
129
130
131
132
133
134
135
136
137
138
139
140
141
142
143
144
145
146
147
148
149
150
151
152
153
154
155
156
157
158
159
160
161
162
163
164
165
166
167
168
169
170
171
172
173
174
175
176
177
177
178
179
180
181
182
183
184
185
186
187
188
189
190
191
192
193
194
195
196
197
198
199
200
201
202
203
204
205
206
207
208
209
209
210
211
212
213
214
215
216
216
217
218
219
220
221
222
223
224
225
226
227
228
229
230
231
232
233
234
235
236
237
238
239
240
241
242
243
244
245
246
247
248
249
250
251
252
253
254
255
256
257
258
259
260
261
262
263
264
265
266
267
268
269
270


In [19]:

df=pd.DataFrame({"Название": title , "Цена": price, "Страна производства": country, "Бренд": brand, "Тип установки": setting, "Тип подключения": connection, "Тип кофе": coffee, "Объем бойлера, л": boiler_volume, "Кофемолка": grinder, "Капуччинатор": frother})

In [7]:
for i in range(len(brand)):
  brand[i]=brand[i].split('  ')[0]


In [8]:
df

Название          Цена  \
0    Кофемашина рожковая MANIFESTA Professionale 2 ...  349 490 руб.   
1    Кофемашина рожковая C.M.A. ASTORIA Pratic Avan...  480 000 руб.   
2    Кофемашина рожковая NUOVA SIMONELLI Appia Life...  352 314 руб.   
3    Кофемашина рожковая QUALITY ESPRESSO FUTURMAT ...  226 333 руб.   
4    Кофемашина рожковая ESTREMO Djua 2 высокие гру...  239 250 руб.   
..                                                 ...           ...   
573  Кофемашина рожковая LA PAVONI GENIALE3VB 3 низ...  644 080 руб.   
574  Кофемашина рожковая LA PAVONI DESIDERIO3VB 3 н...  869 120 руб.   
575  Кофемашина рожковая CREM EX2 Mini 1 высокая гр...  243 500 руб.   
576  Кофемашина рожковая LA PAVONI DIA3SV 3 низкие ...  826 440 руб.   
577  Кофемашина рожковая CREM EX2 2 высокие группы,...  322 800 руб.   

    Страна производства                                              Бренд  \
0                Италия  MANIFESTA                                     ...   
1                Италия  C.M.A. Astoria                                ...   
2                Италия  Nuova Simonelli                               ...   
3               Испания  QUALITY ESPRESSO FUTURMAT                     ...   
4                 Китай  ESTREMO                                       ...   
..                  ...                                                ...   
573           Швейцария  LA PAVONI                                     ...   
574           Швейцария  LA PAVONI                                     ...   
575             Испания  CREM                                          ...   
576           Швейцария  LA PAVONI                                     ...   
577             Испания  CREM                                          ...   

    Тип установки Тип подключения Тип кофе Объем бойлера, л Кофемолка  \
0      Настольная   Электрическое  молотый             11.5       Нет   
1      Настольная   Электрическое  молотый             10.5       Нет   
2      Настольная   Электрическое  молотый               11       Нет   
3      Настольная   Электрическое  молотый                5       Нет   
4      Настольная   Электрическое  молотый               11       Нет   
..            ...             ...      ...              ...       ...   
573    Настольная   Электрическое  молотый               21       Нет   
574    Настольная   Электрическое  молотый               21       Нет   
575    Настольная   Электрическое  молотый                6       Нет   
576    Настольная   Электрическое  молотый             22.5       Нет   
577    Настольная   Электрическое  молотый             11.5       Нет   

    Капуччинатор  
0           Есть  
1           Есть  
2           Есть  
3           Есть  
4           Есть  
..           ...  
573         Есть  
574         Есть  
575         Есть  
576         Есть  
577         Есть  

[578 rows x 10 columns]

In [26]:
df['Цена'] = (df['Цена'].apply(lambda item: ''.join(filter(str.isdigit, item))))
df['Цена'] = pd.to_numeric(df['Цена'], errors='coerce')

In [ ]:
df['Цена'].dtype

dtype('int64')

In [ ]:
set(df["Страна производства"])

{'Германия',
 'Испания',
 'Италия',
 'Китай',
 'Португалия',
 'Россия',
 'США',
 'Турция',
 'Франция',
 'Швейцария',
 '—-'}

In [10]:
countries_counts=df['Страна производства'].value_counts().to_frame()
countries_counts=countries_counts.drop(index='—-')

print(countries_counts.index)
countries = ['Italy', 'Switzerland', 'Spain', 'Portugal', 'Russia', 'China',
             'Turkey', 'France', 'USA', 'Germany']
countries_counts.index = countries


Index(['Италия', 'Швейцария', 'Испания', 'Португалия', 'Россия', 'Китай',
       'Турция', 'Франция', 'США', 'Германия'],
      dtype='object', name='Страна производства')


In [11]:
countries_counts

count
Italy          425
Switzerland     40
Spain           38
Portugal        25
Russia          14
China           13
Turkey          10
France           5
USA              4
Germany          2

In [12]:
countries_counts["Страны"]=['Италия', 'Швейцария', 'Испания', 'Португалия', 'Россия', 'Китай',
       'Турция', 'Франция', 'США', 'Германия']

In [13]:
countries_counts

count      Страны
Italy          425      Италия
Switzerland     40   Швейцария
Spain           38     Испания
Portugal        25  Португалия
Russia          14      Россия
China           13       Китай
Turkey          10      Турция
France           5     Франция
USA              4         США
Germany          2    Германия

In [14]:
df

Название    Цена  \
0    Кофемашина рожковая MANIFESTA Professionale 2 ...  349490   
1    Кофемашина рожковая C.M.A. ASTORIA Pratic Avan...  480000   
2    Кофемашина рожковая NUOVA SIMONELLI Appia Life...  352314   
3    Кофемашина рожковая QUALITY ESPRESSO FUTURMAT ...  226333   
4    Кофемашина рожковая ESTREMO Djua 2 высокие гру...  239250   
..                                                 ...     ...   
573  Кофемашина рожковая LA PAVONI GENIALE3VB 3 низ...  644080   
574  Кофемашина рожковая LA PAVONI DESIDERIO3VB 3 н...  869120   
575  Кофемашина рожковая CREM EX2 Mini 1 высокая гр...  243500   
576  Кофемашина рожковая LA PAVONI DIA3SV 3 низкие ...  826440   
577  Кофемашина рожковая CREM EX2 2 высокие группы,...  322800   

    Страна производства                                              Бренд  \
0                Италия  MANIFESTA                                     ...   
1                Италия  C.M.A. Astoria                                ...   
2                Италия  Nuova Simonelli                               ...   
3               Испания  QUALITY ESPRESSO FUTURMAT                     ...   
4                 Китай  ESTREMO                                       ...   
..                  ...                                                ...   
573           Швейцария  LA PAVONI                                     ...   
574           Швейцария  LA PAVONI                                     ...   
575             Испания  CREM                                          ...   
576           Швейцария  LA PAVONI                                     ...   
577             Испания  CREM                                          ...   

    Тип установки Тип подключения Тип кофе Объем бойлера, л Кофемолка  \
0      Настольная   Электрическое  молотый             11.5       Нет   
1      Настольная   Электрическое  молотый             10.5       Нет   
2      Настольная   Электрическое  молотый               11       Нет   
3      Настольная   Электрическое  молотый                5       Нет   
4      Настольная   Электрическое  молотый               11       Нет   
..            ...             ...      ...              ...       ...   
573    Настольная   Электрическое  молотый               21       Нет   
574    Настольная   Электрическое  молотый               21       Нет   
575    Настольная   Электрическое  молотый                6       Нет   
576    Настольная   Электрическое  молотый             22.5       Нет   
577    Настольная   Электрическое  молотый             11.5       Нет   

    Капуччинатор  
0           Есть  
1           Есть  
2           Есть  
3           Есть  
4           Есть  
..           ...  
573         Есть  
574         Есть  
575         Есть  
576         Есть  
577         Есть  

[578 rows x 10 columns]

In [18]:
for i in range(len(brand)):
  brand[i]=brand[i].split('  ')[0]
print(set(brand))

{'ELEKTRA', 'RANCILIO', 'DALLA CORTE', 'WMF', 'ROBUSTEZZA', 'EMPERO', 'KIMNELLI', 'AUTOMATIC COFFEE MACHINES SRL', 'GUFO', 'CASADIO', 'C.M.A. Astoria', 'CREM', 'BEZZERA', 'CONTI', 'CIME', 'GEMILAI', 'NEW COFFEE MACHINES', 'LA CIMBALI', 'ESTREMO', 'Victoria Arduino', 'FIAMMA', 'COFFF', 'MANIFESTA', 'HEYLO', 'SYNESSO', 'LA PAVONI', 'Nuova Simonelli', 'SANREMO', 'MAVAM', 'CARIMALI', 'DAZHENG', 'QUALITY ESPRESSO FUTURMAT'}


In [21]:
brand_count=df['Бренд'].value_counts().to_frame()

In [22]:
brand_count=brand_count.reset_index()
display(brand_count)

Бренд  count
0                Victoria Arduino     89
1                 Nuova Simonelli     51
2                        RANCILIO     43
3                     DALLA CORTE     41
4                       LA PAVONI     40
5                         SANREMO     38
6                            CREM     35
7                      ROBUSTEZZA     33
8                        CARIMALI     28
9                          FIAMMA     25
10                 C.M.A. Astoria     18
11                        BEZZERA     18
12                        ELEKTRA     17
13                           CIME     13
14                     LA CIMBALI     10
15                         EMPERO     10
16                          COFFF      9
17  AUTOMATIC COFFEE MACHINES SRL      9
18            NEW COFFEE MACHINES      8
19                           GUFO      6
20                       KIMNELLI      6
21                          CONTI      5
22                      MANIFESTA      5
23                        DAZHENG      4
24                        CASADIO      3
25      QUALITY ESPRESSO FUTURMAT      3
26                          MAVAM      3
27                        ESTREMO      2
28                          HEYLO      2
29                            WMF      2
30                        GEMILAI      1
31                        SYNESSO      1

In [23]:
brand_count.loc[brand_count['count']<6, "Бренд" ]="Другие бренды"

In [24]:
display(brand_count)

Бренд  count
0                Victoria Arduino     89
1                 Nuova Simonelli     51
2                        RANCILIO     43
3                     DALLA CORTE     41
4                       LA PAVONI     40
5                         SANREMO     38
6                            CREM     35
7                      ROBUSTEZZA     33
8                        CARIMALI     28
9                          FIAMMA     25
10                 C.M.A. Astoria     18
11                        BEZZERA     18
12                        ELEKTRA     17
13                           CIME     13
14                     LA CIMBALI     10
15                         EMPERO     10
16                          COFFF      9
17  AUTOMATIC COFFEE MACHINES SRL      9
18            NEW COFFEE MACHINES      8
19                           GUFO      6
20                       KIMNELLI      6
21                  Другие бренды      5
22                  Другие бренды      5
23                  Другие бренды      4
24                  Другие бренды      3
25                  Другие бренды      3
26                  Другие бренды      3
27                  Другие бренды      2
28                  Другие бренды      2
29                  Другие бренды      2
30                  Другие бренды      1
31                  Другие бренды      1

In [27]:
print(df["Цена"])

0      349490
1      480000
2      352314
3      226333
4      239250
        ...  
573    644080
574    869120
575    243500
576    826440
577    322800
Name: Цена, Length: 578, dtype: int64


In [28]:
grouped_machines1 = df.groupby('Страна производства')
result = grouped_machines1.agg(
    num_brand=('Бренд', 'nunique'),
    top_brand=('Бренд', lambda x: x.mode()[0]),
    all_brand=('Бренд', lambda x: ', '.join(x.unique())),
    avg_price=('Цена', 'mean')
).reset_index()
display(result)

Страна производства  num_brand            top_brand  \
0             Германия          1                  WMF   
1              Испания          2                 CREM   
2               Италия         16     Victoria Arduino   
3                Китай          4             KIMNELLI   
4           Португалия          1               FIAMMA   
5               Россия          2  NEW COFFEE MACHINES   
6                  США          2                MAVAM   
7               Турция          1               EMPERO   
8              Франция          1                CONTI   
9            Швейцария          1            LA PAVONI   
10                  —-          1                HEYLO   

                                            all_brand     avg_price  
0                                                 WMF  1.435126e+06  
1                     QUALITY ESPRESSO FUTURMAT, CREM  4.523202e+05  
2   MANIFESTA, C.M.A. Astoria, Nuova Simonelli, RO...  8.843709e+05  
3                 ESTREMO, DAZHENG, KIMNELLI, GEMILAI  3.119910e+05  
4                                              FIAMMA  3.694188e+05  
5                           GUFO, NEW COFFEE MACHINES  4.432321e+05  
6                                      SYNESSO, MAVAM  2.240282e+06  
7                                              EMPERO  2.752634e+05  
8                                               CONTI  6.027972e+05  
9                                           LA PAVONI  4.879491e+05  
10                                              HEYLO  7.353500e+05

In [29]:
fig = px.pie(brand_count,
             values='count',
             hole=0.4,
             names='Бренд',
             hover_data=['count'],
             color_discrete_sequence=px.colors.sequential.Magma,
             labels={
                 'count': 'количество',
                 'Бренд': 'бренд кофемашины'
             },
             title='Распределение кофемашин по брендам',
             width=1000,
             height=600
            )


fig.show()

In [131]:
from dash import Dash, dcc, html, Input, Output
import plotly.express as px
import pandas as pd
from sklearn.datasets import load_iris
import dash_bootstrap_components as dbc
from dash import dcc
import dash_core_components as dcc
app = Dash(__name__)

fig11 = px.histogram(df, x=df['Цена'], nbins=100, title='Гистограмма цен на кофемашины', color_discrete_sequence=["#636EFA"])
app.layout = html.Div([
    dcc.Graph(
        id='price_histogram',
        figure=fig
    )
])

if __name__ == '__main__':

  app.run(debug=True)

<IPython.core.display.Javascript object>

In [31]:

app = Dash(__name__)

fig2 = px.choropleth(countries_counts,
                    locations=countries_counts.index,
                    locationmode="country names",
                    color=countries_counts["count"],
                    color_continuous_scale="plasma",
                    title="Количество наблюдений по странам",
                    projection='natural earth',

                    width=800,
                    height=500)

app.layout = html.Div([
    html.H1("Количество наблюдений по странам"),
    dcc.Graph(
        id='world_map',
        figure=fig2
    )
])
if __name__ == '__main__':

  app.run(debug=True)

<IPython.core.display.Javascript object>

In [32]:
top_countries = countries_counts.nlargest(5, 'count')["Страны"].tolist()

In [33]:
top_countries

['Италия', 'Швейцария', 'Испания', 'Португалия', 'Россия']

In [132]:

app = dash.Dash(__name__, external_stylesheets=[dbc.themes.BOOTSTRAP])
dashboard_layout = dbc.Container([
    html.H1("Дашборд с информацией про кофемашины", className='text-center text-primary mb-4'),
    html.H3("Распределение кофемашин по странам", className='text-center mb-2'),
    dbc.Row([
        dbc.Col(dcc.Graph(id='world-map', figure=fig2), width=12)
    ], className='mb-4'),
    html.H3("Гистограмма цен на кофемашины", className='text-center mb-2'),
    dbc.Row([
        dbc.Col(dcc.Graph(id='price_histogram', figure=fig11), width=12)
    ], className='mb-4'),
    dbc.Row([
        dbc.Col([
            dbc.Card([
                dbc.CardHeader('Выберите страну'),
                dbc.CardBody([
                    dcc.Dropdown(
                        id='country-dropdown',
                        options=[{'label': country, 'value': country} for country in top_countries],
                        placeholder='Выберите страну',
                    )
                ])
            ])
        ], width=4),
        dbc.Col([
            dbc.Card([
                dbc.CardHeader('Информация о кофемашинах'),
                dbc.CardBody([
                    html.P(id="top_brand", className="card-text"),
                    html.P(id="avg_price", className="card-text"),
                    html.P(id="num_brand", className="card-text"),
                    html.P(id='all_brand', className="card-text")
                ])
            ])
        ], width=8)
    ])
], fluid=True)

@app.callback(
    [Output("top_brand", "children"),
     Output("avg_price", "children"),
     Output("num_brand", "children"),
     Output("all_brand", "children")],
    [Input("country-dropdown", "value")]
)
def update_info(selected_country):
    if not selected_country:
        return "Выберите страну", "", "", ""

    filtered_data = result[result["Страна производства"] == selected_country]

    if filtered_data.empty:
        print("AAAAAA")
        return "Нет данных", "Нет данных", "Нет данных", "Нет данных"

    row = result[result["Страна производства"] == selected_country].iloc[0]
    return (
        f"Топовый бренд: {row['top_brand']}",
        f"Средняя цена: {row['avg_price']} руб.",
        f"Количество брендов: {row['num_brand']}",
        f"Все бренды: {row['all_brand']}"
    )


app.layout = dashboard_layout

if __name__ == '__main__':
    app.run(debug=True)


<IPython.core.display.Javascript object>

In [35]:
spaces=[]
links_s=[]
for i in range (1, 55):
  print(i)
  html34 = requests.get("https://www.cian.ru/cat.php?deal_type=rent&engine_version=2&offer_type=offices&office_type%5B0%5D=4&p="+str(i)+"&region=1").content.decode('utf-8')
  #print(html34)
  soup = BeautifulSoup(html34, 'html.parser')
  spaces.extend(soup.find_all("div", class_="_32bbee5fda--offer-container--Zhu18"))
  for a in soup.find_all('a', class_="_32bbee5fda--header-link--RFvxs"):
        links_s.append(a['href'])
print(len(spaces))


1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
1509


'for a in soup.find_all(\'a\', class_="_32bbee5fda--header-link--RFvxs"):\n  links_s.append(a[\'href\'])\nfor product in products:\n  links.append("https://r-komplekt.ru" + product.find("a")["href"])'

In [ ]:
display(len(links_s))

1523

In [36]:
adress = []
metro = []
prices = []
area=[]
i=1
for current_link in links_s:
    print(i)
    html5 = requests.get(current_link).content.decode('utf-8')
    soup = BeautifulSoup(html5, 'html.parser')
    station = soup.find('a', class_="a10a3f92e9--underground_link--VnUVj")
    if station is not None:
        metro.append(station.get_text(strip=True))
    else:
        continue
    price=soup.find("div", class_="a10a3f92e9--amount--ON6i1")
    if price is not None:
        prices.append(price.get_text(strip=True))
    else:
        continue
    adress0=soup.find_all('a', class_="a10a3f92e9--address--SMU25")
    if adress0 is not None:
      adress_list = [a.get_text(strip=True) for a in adress0]
      adress.append(', '.join(adress_list))
    else:
      adress.append(None)
    area0=soup.find("span", class_="a10a3f92e9--color_text-primary-default--vSRPB a10a3f92e9--lineHeight_6u--cedXD a10a3f92e9--fontWeight_bold--BbhnX a10a3f92e9--fontSize_16px--QNYmt a10a3f92e9--display_block--KYb25 a10a3f92e9--text--e4SBY")
    if area0 is not None:
        area.append(area0.get_text(strip=True))
    else:
        continue
    i+=1

print(prices)





1
2
3
4
5
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56
57
58
59
60
61
62
63
64
65
66
67
68
69
70
71
71
72
73
74
75
76
77
78
79
80
81
82
83
84
85
86
87
88
89
90
91
92
93
94
95
96
97
98
99
100
101
102
103
104
105
106
107
108
109
110
111
112
112
113
114
115
116
117
118
119
120
121
122
123
124
125
126
127
128
129
130
131
132
133
134
135
136
137
138
139
140
141
142
143
144
145
146
147
148
149
150
151
152
153
154
155
156
157
158
159
160
161
162
163
164
165
166
167
168
169
170
171
172
173
174
175
176
177
178
179
180
181
182
183
184
185
186
187
188
189
190
191
192
193
194
195
196
197
198
199
200
201
202
203
204
205
206
207
208
209
210
211
212
213
214
215
216
217
218
219
220
221
222
223
224
225
226
227
228
229
230
231
232
233
234
235
236
237
238
239
240
241
242
243
244
245
246
247
248
249
250
251
252
253
254
255
256
257
258
259
260
261
262
263
264
265
266
267
268
269
269
270
271
271
272
273

In [56]:
dfs=pd.DataFrame({"Адрес": adress, "Ближайшее метро": metro, "Цена": prices, "Площадь": area})

df=pd.DataFrame({"Название": title , "Цена": price, "Страна производства": country, "Бренд": brand, "Тип установки": setting, "Тип подключения": connection, "Тип кофе": coffee, "Объем бойлера, л": boiler_volume, "Кофемолка": grinder, "Капуччинатор": frother})



In [57]:
display(dfs)

Адрес  Ближайшее метро  \
0     Москва, САО, р-н Савеловский, ул. Верхняя Масл...  Петровский Парк   
1     Москва, САО, р-н Савеловский, ул. Верхняя Масл...  Петровский Парк   
2     Москва, ЗАО, р-н Тропарево-Никулино, просп. Ве...     Юго-Западная   
3     Москва, ЗАО, р-н Тропарево-Никулино, просп. Ве...     Юго-Западная   
4     Москва, САО, р-н Западное Дегунино, Пяловская ...      Селигерская   
...                                                 ...              ...   
1470  Москва, ЦАО, р-н Замоскворечье, Садовническая ...    Новокузнецкая   
1471           Москва, ЦАО, р-н Якиманка, Житная ул., 4     Добрынинская   
1472  Москва, ЗАО, р-н Тропарево-Никулино, просп. Ве...     Юго-Западная   
1473  Москва, ЗАО, р-н Тропарево-Никулино, просп. Ве...     Юго-Западная   
1474  Москва, СЗАО, р-н Покровское-Стрешнево, Волоко...          Спартак   

                           Цена         Площадь  
0      300 000 – 450 000 ₽/мес.  46,4 – 80,6 м²  
1      300 000 – 450 000 ₽/мес.  46,4 – 80,6 м²  
2       10 400 – 532 000 ₽/мес.  5,2 – 319,2 м²  
3       10 400 – 532 000 ₽/мес.  5,2 – 319,2 м²  
4     99 800 – 4 275 000 ₽/мес.   28 – 4 500 м²  
...                         ...             ...  
1470           2 500 000 ₽/мес.          750 м²  
1471           4 100 000 ₽/мес.        585,5 м²  
1472    10 400 – 532 000 ₽/мес.  5,2 – 319,2 м²  
1473    10 400 – 532 000 ₽/мес.  5,2 – 319,2 м²  
1474             250 000 ₽/мес.           50 м²  

[1475 rows x 4 columns]

In [39]:
import re

In [115]:
districts=[]
okruga=[]
for ad in adress:
  parts = ad.split(',')
  start_index = ad.find("р-н")
  start_index += len("р-н") + 1
  end_index = ad.find(",", start_index)
  if end_index == -1:
      end_index = len(ad)
  districts.append(ad[start_index:end_index].strip())
  okruga.append(parts[1].strip())


pattern = r'(?:\bр\s+|\bквартал\s+)([^,]+)'

dfs['Район3'] = dfs['Адрес'].apply(
    lambda addr: (m.group(1).strip()
                  if (m := re.search(pattern, addr, flags=re.IGNORECASE))
                  else None)
)
#dfs.insert(len(dfs.columns), 'Район', districts)
#dfs.insert(len(dfs.columns), 'Округ', okruga)





In [116]:
display(dfs)

Адрес Ближайшее метро  \
6     Москва, САО, р-н Хорошевский, 1-я Магистральна...    Полежаевская   
7     Москва, СЗАО, р-н Хорошево-Мневники, 3-я Хорош...           Зорге   
8     Москва, СЗАО, р-н Хорошево-Мневники, 3-я Хорош...           Зорге   
9     Москва, СЗАО, р-н Хорошево-Мневники, 3-я Хорош...           Зорге   
10    Москва, ЗАО, р-н Можайский, Верейская 41 жилой...       Давыдково   
...                                                 ...             ...   
1464  Москва, ЮВАО, р-н Лефортово, проезд Завода Сер...    Авиамоторная   
1466  Москва, ЗАО, р-н Фили-Давыдково, Большая Филев...      Пионерская   
1469    Москва, САО, р-н Савеловский, Вятская ул., 27С7     Савёловская   
1471           Москва, ЦАО, р-н Якиманка, Житная ул., 4    Добрынинская   
1474  Москва, СЗАО, р-н Покровское-Стрешнево, Волоко...         Спартак   

                  Цена   Площадь                 Район Округ  Чистая цена  \
6       299 389 ₽/мес.   80,7 м²           Хорошевский   САО       299389   
7       544 335 ₽/мес.  153,1 м²     Хорошево-Мневники  СЗАО       544335   
8     1 980 718 ₽/мес.  696,7 м²     Хорошево-Мневники  СЗАО      1980718   
9       681 462 ₽/мес.  196,2 м²     Хорошево-Мневники  СЗАО       681462   
10      388 316 ₽/мес.  104,5 м²             Можайский   ЗАО       388316   
...                ...       ...                   ...   ...          ...   
1464    400 000 ₽/мес.    230 м²             Лефортово  ЮВАО       400000   
1466    400 000 ₽/мес.    250 м²        Фили-Давыдково   ЗАО       400000   
1469  1 257 600 ₽/мес.    262 м²           Савеловский   САО      1257600   
1471  4 100 000 ₽/мес.  585,5 м²              Якиманка   ЦАО      4100000   
1474    250 000 ₽/мес.     50 м²  Покровское-Стрешнево  СЗАО       250000   

      Чистая площадь    Цена за м Район3  
6               80.7  3709.900867   None  
7              153.1  3555.421293   None  
8              696.7  2842.999856   None  
9              196.2  3473.302752   None  
10             104.5  3715.942584   None  
...              ...          ...    ...  
1464           230.0  1739.130435   None  
1466           250.0  1600.000000   None  
1469           262.0  4800.000000   None  
1471           585.5  7002.561913   None  
1474            50.0  5000.000000   None  

[1350 rows x 10 columns]

In [59]:
prices_cl=[]
for price in prices:
  cleaned_price_str =price.replace('\xa0', '').replace('₽/мес.', '')
  prices_cl.append(cleaned_price_str)



In [60]:
dfs.insert(len(dfs.columns), 'Чистая цена', prices_cl)

In [113]:
display(dfs)

Адрес Ближайшее метро  \
6     Москва, САО, р-н Хорошевский, 1-я Магистральна...    Полежаевская   
7     Москва, СЗАО, р-н Хорошево-Мневники, 3-я Хорош...           Зорге   
8     Москва, СЗАО, р-н Хорошево-Мневники, 3-я Хорош...           Зорге   
9     Москва, СЗАО, р-н Хорошево-Мневники, 3-я Хорош...           Зорге   
10    Москва, ЗАО, р-н Можайский, Верейская 41 жилой...       Давыдково   
...                                                 ...             ...   
1464  Москва, ЮВАО, р-н Лефортово, проезд Завода Сер...    Авиамоторная   
1466  Москва, ЗАО, р-н Фили-Давыдково, Большая Филев...      Пионерская   
1469    Москва, САО, р-н Савеловский, Вятская ул., 27С7     Савёловская   
1471           Москва, ЦАО, р-н Якиманка, Житная ул., 4    Добрынинская   
1474  Москва, СЗАО, р-н Покровское-Стрешнево, Волоко...         Спартак   

                  Цена   Площадь                 Район Округ  Чистая цена  \
6       299 389 ₽/мес.   80,7 м²           Хорошевский   САО       299389   
7       544 335 ₽/мес.  153,1 м²     Хорошево-Мневники  СЗАО       544335   
8     1 980 718 ₽/мес.  696,7 м²     Хорошево-Мневники  СЗАО      1980718   
9       681 462 ₽/мес.  196,2 м²     Хорошево-Мневники  СЗАО       681462   
10      388 316 ₽/мес.  104,5 м²             Можайский   ЗАО       388316   
...                ...       ...                   ...   ...          ...   
1464    400 000 ₽/мес.    230 м²             Лефортово  ЮВАО       400000   
1466    400 000 ₽/мес.    250 м²        Фили-Давыдково   ЗАО       400000   
1469  1 257 600 ₽/мес.    262 м²           Савеловский   САО      1257600   
1471  4 100 000 ₽/мес.  585,5 м²              Якиманка   ЦАО      4100000   
1474    250 000 ₽/мес.     50 м²  Покровское-Стрешнево  СЗАО       250000   

      Чистая площадь    Цена за м  
6               80.7  3709.900867  
7              153.1  3555.421293  
8              696.7  2842.999856  
9              196.2  3473.302752  
10             104.5  3715.942584  
...              ...          ...  
1464           230.0  1739.130435  
1466           250.0  1600.000000  
1469           262.0  4800.000000  
1471           585.5  7002.561913  
1474            50.0  5000.000000  

[1350 rows x 9 columns]

In [ ]:
print(area)

['46,4 – 80,6 м²', '46,4 – 80,6 м²', '5,2 – 319,2 м²', '5,2 – 319,2 м²', '51 – 291,4 м²', '51 – 291,4 м²', '28 – 4\xa0500 м²', '28 – 4\xa0500 м²', '80,7\xa0м²', '153,1\xa0м²', '696,7\xa0м²', '196,2\xa0м²', '104,5\xa0м²', '210,2\xa0м²', '114,1\xa0м²', '1\xa0433,4\xa0м²', '623\xa0м²', '96\xa0м²', '87,3\xa0м²', '141,3\xa0м²', '137,14\xa0м²', '128,94\xa0м²', '83,56\xa0м²', '80,2\xa0м²', '228,87\xa0м²', '128,3\xa0м²', '81,22\xa0м²', '83,05\xa0м²', '66,54\xa0м²', '114,7\xa0м²', '57,7\xa0м²']


In [61]:
area_cl=[]
for area0 in area:
  cleaned_area_str =area0.replace('\xa0', '').replace('м²', '')
  area_cl.append(cleaned_area_str)

In [62]:
dfs.insert(len(dfs.columns), 'Чистая площадь', area_cl)

In [63]:
dfs = dfs[~dfs['Цена'].str.contains('–')]

In [64]:
dfs = dfs[~dfs['Чистая площадь'].str.contains('–')]

In [65]:
dfs['Чистая площадь'] = dfs['Чистая площадь'].apply(lambda x: float(x.replace(',', '.')))

In [66]:
dfs['Чистая цена'] = (dfs['Чистая цена'].apply(lambda item: ''.join(filter(str.isdigit, item))))
dfs['Чистая цена'] = pd.to_numeric(dfs['Чистая цена'], errors='coerce')

In [67]:
print(dfs['Чистая цена'].dtypes)

int64


In [68]:
dfs["Цена за м"]=dfs["Чистая цена"]/dfs["Чистая площадь"]

In [69]:
print(dfs['Чистая площадь'])

6        80.7
7       153.1
8       696.7
9       196.2
10      104.5
        ...  
1466    250.0
1469    262.0
1470    750.0
1471    585.5
1474     50.0
Name: Чистая площадь, Length: 1387, dtype: float64


In [107]:
dfs = dfs[dfs['Чистая площадь']<700]
dfs = dfs[dfs['Цена за м']<25000]

In [135]:
import dash
from dash import dcc, html
import pandas as pd
import plotly.express as px

app = dash.Dash(__name__)

fig4 = px.histogram(dfs, x=dfs['Чистая площадь'], nbins=150, title='Гистограмма площадей под общепиты', color_discrete_sequence=["#636EFA"])
app.layout = html.Div([
    dcc.Graph(
        id='price_histogram',
        figure=fig4
    )
])
fig4.update_layout(
    yaxis_title='Количество объявлений',
    xaxis_title='Площадь помещения '
)

if __name__ == '__main__':
  app.run(debug=True)


<IPython.core.display.Javascript object>

In [126]:
app = dash.Dash(__name__)

fig5 = px.histogram(dfs, x='Цена за м', nbins=50, title='Распределение цены за кв. метр', color_discrete_sequence=["#636EFA"], labels={
        'Цена за м': 'Цена за квадратный метр (₽)',
        'count': 'Количество объявлений'
    })
app.layout = html.Div([
    dcc.Graph(
        id='price_histogram',
        figure=fig5
    )
])
fig5.update_layout(
    yaxis_title='Количество объявлений',
    xaxis_title='Цена за квадратный метр (₽)'
)

if __name__ == '__main__':
  app.run(debug=True)

<IPython.core.display.Javascript object>

In [118]:
app = dash.Dash(__name__, external_stylesheets=[dbc.themes.BOOTSTRAP])
district_count = dfs['Район'].value_counts().reset_index()
district_count.columns = ['Район', 'Количество']
district_count = district_count[district_count['Район'] != 'ква']
top_10 = district_count.head(10)
other_count = district_count['Количество'][10:].sum()
other_row = pd.DataFrame({'Район': ['Другие'], 'Количество': [other_count]})
top_10 = pd.concat([top_10, other_row], ignore_index=True)
fig7 = px.pie(
    top_10,
    values='Количество',
    names='Район',
    hole=0.4,
    hover_data=['Количество'],
    color_discrete_sequence=px.colors.sequential.Magma,
    labels={
        'Количество': 'Количество объявлений',
        'Район': 'Район Москвы'
    },
    title='📍 Распределение объявлений по районам Москвы',
    width=1000,
    height=600
)

fig7.update_traces(
    textinfo='percent+label',
    pull=[0.05]*len(district_count),
    marker=dict(line=dict(color='#FFFFFF', width=2))
)
app.layout = html.Div([
    dcc.Graph(
        id='price_histogram',
        figure=fig7
    )
])


if __name__ == '__main__':
    app.run(debug=True)

<IPython.core.display.Javascript object>

In [74]:
from google.colab import files
files.upload()
dfm = pd.read_csv('list_of_moscow_metro_stations.csv')
print(dfm.head())

Saving list_of_moscow_metro_stations.csv to list_of_moscow_metro_stations.csv
          Line LineColor               Name   Latitude  Longitude  Order
0  Калининская    FFCD1C         Новокосино  55.745113  37.864052      0
1  Калининская    FFCD1C        Новогиреево  55.752237  37.814587      1
2  Калининская    FFCD1C             Перово  55.750980  37.784220      2
3  Калининская    FFCD1C  Шоссе энтузиастов  55.758090  37.751703      3
4  Калининская    FFCD1C       Авиамоторная  55.751933  37.717444      4


In [84]:
counts = dfs.groupby('Ближайшее метро').size().reset_index(name='count')
merged = dfm.merge(counts, left_on='Name', right_on='Ближайшее метро', how='left')
merged['count'] = merged['count'].fillna(0)
fig9 = px.scatter_mapbox(merged,
    lat='Latitude',
    lon='Longitude',
    hover_name='Name',
    color='count',
    size='count',
    zoom=10,
    height=600,
    mapbox_style='open-street-map'
)
fig9.show()

In [80]:
display(merged)

Line LineColor                      Name   Latitude  Longitude  \
0    Калининская    FFCD1C                Новокосино  55.745113  37.864052   
1    Калининская    FFCD1C               Новогиреево  55.752237  37.814587   
2    Калининская    FFCD1C                    Перово  55.750980  37.784220   
3    Калининская    FFCD1C         Шоссе энтузиастов  55.758090  37.751703   
4    Калининская    FFCD1C              Авиамоторная  55.751933  37.717444   
..           ...       ...                       ...        ...        ...   
235    Монорельс    006DA8         Улица Милашенкова  55.821876  37.591206   
236    Монорельс    006DA8                 Телецентр  55.821795  37.608975   
237    Монорельс    006DA8  Улица Академика Королёва  55.821821  37.627167   
238    Монорельс    006DA8         Выставочный центр  55.824086  37.638494   
239    Монорельс    006DA8  Улица Сергея Эйзенштейна  55.829305  37.644997   

     Order    Ближайшее метро  count  
0        0         Новокосино    6.0  
1        1        Новогиреево    4.0  
2        2             Перово    4.0  
3        3                NaN    0.0  
4        4       Авиамоторная    5.0  
..     ...                ...    ...  
235      1                NaN    0.0  
236      2                NaN    0.0  
237      3                NaN    0.0  
238      4  Выставочный центр    1.0  
239      5                NaN    0.0  

[240 rows x 8 columns]

In [136]:
import dash
from dash import dcc, html
import dash_bootstrap_components as dbc
from dash import Dash

app = Dash(__name__, external_stylesheets=[dbc.themes.BOOTSTRAP])

hist_area = dbc.Card([
    dbc.CardHeader(html.H4('🏠 Распределение площадей по Москве', className='card-title')),
    dbc.CardBody([
                    dcc.Graph(figure=fig4)
                ])
])

hist_moscow = dbc.Card([
    dbc.CardHeader(html.H4('💰 Распределение цен по Москве', className='card-title')),
    dbc.CardBody([
        dcc.Graph(figure=fig5)
    ])
])

dropdown_and_district_hist = dbc.Row([
    dbc.Col([
            html.Label('🏙️ Выберите округ:', className='form-label fw-bold mb-2'),
            dcc.Dropdown(
                id='okrug-dropdown',
                options=[{'label': o, 'value': o} for o in set(okruga)],
                placeholder='Выберите округ'
            )
        ], width=4),
    dbc.Col([
        dbc.Card([
            dbc.CardHeader(html.H4('📊 Распределение цен по выбранному округу', className='card-title')),
            dbc.CardBody([
                dcc.Graph(id='histogram-district')
            ])
        ])
    ], md=8)
])

pie_chart = dbc.Card([
    dbc.CardHeader(html.H4('📍 Распределение объявлений по районам', className='card-title')),
    dbc.CardBody([
        dcc.Graph(figure=fig7)
    ])
])

map_chart = dbc.Card([
    dbc.CardHeader(html.H4('🗺️ Карта объявлений по Москве', className='card-title')),
    dbc.CardBody([
        dcc.Graph(figure=fig9)
    ])
])

# Layout приложения
app.layout = dbc.Container([
    dbc.Row([
        dbc.Col(hist_area, width=12)
    ], className='mb-4'),

    dbc.Row([
        dbc.Col(hist_moscow, width=12)
    ], className='mb-4'),

    dropdown_and_district_hist,

    dbc.Row([
        dbc.Col(pie_chart, width=12)
    ], className='my-4'),

    dbc.Row([
        dbc.Col(map_chart, width=12)
    ])
], fluid=True)

@app.callback(
    Output('histogram-district', 'figure'),
    Input('okrug-dropdown', 'value')
)
def update_histogram(selected_okrug):
    if selected_okrug is None or selected_okrug not in dfs['Округ'].values:
        return px.histogram(title='Выберите округ для отображения данных')

    filtered_df = dfs[dfs['Округ'] == selected_okrug]
    fig8 = px.histogram(
        filtered_df,
        x='Цена за м',
        nbins=50,
        title=f'💰 Цена за м² в округе {selected_okrug}',
        color_discrete_sequence=['#00CC96']
    )
    fig8.update_layout(xaxis_title='Цена за м²', yaxis_title='Количество объектов')
    return fig8

if __name__ == '__main__':
    app.run(debug=True)

<IPython.core.display.Javascript object>

In [10]:
html100 = requests.get('https://ru.wikipedia.org/wiki/%D0%A1%D0%BF%D0%B8%D1%81%D0%BE%D0%BA_%D1%81%D1%82%D0%B0%D0%BD%D1%86%D0%B8%D0%B9_%D0%9C%D0%BE%D1%81%D0%BA%D0%BE%D0%B2%D1%81%D0%BA%D0%BE%D0%B3%D0%BE_%D0%BC%D0%B5%D1%82%D1%80%D0%BE%D0%BF%D0%BE%D0%BB%D0%B8%D1%82%D0%B5%D0%BD%D0%B0').content.decode('utf-8')
soup = BeautifulSoup(html100, 'html.parser')
stations=soup.find_all('a', class_="mw-kartographer-maplink" )
stations2=soup.find_all('span', class_="sortkey")
st_coords = []
names=[]
for station in stations:
    lat = station.get('data-lat')
    lon = station.get('data-lon')
    if lat and lon:
        st_coords.append({
            'Широта': float(lat),
            'Долгота': float(lon)
        })
for station in stations2:
  name=station.get_text(strip=True)
  names.append(name)
  print(name)
station_coords=pd.DataFrame(st_coords)
display(station_coords)

@1
01
Бульвар Рокоссовского
14
@1
02
Черкизовская
14
@1
03
Преображенская площадь
@1
04
Сокольники
11
D3
@1
05
Красносельская
@1
06
Комсомольская
05
D2
D4
@1
07
Красные Ворота
@1
08
Чистые пруды
06
10
@1
09
Лубянка
07
@1
10
Охотный Ряд
02
03
@1
11
Библиотека имени Ленина
03
04
4А
09
@1
12
Кропоткинская
@1
13
Парк культуры
05
@1
14
Фрунзенская
@1
15
Спортивная
14
@1
16
Воробьёвы горы
@1
17
Университет
@1
18
Проспект Вернадского
11
@1
19
Юго-Западная
@1
20
Тропарёво
@1
21
Румянцево
@1
22
Саларьево
@1
23
@1
24
Прокшино
@1
25
Ольховая
@1
26
Новомосковская
16
@1
26
Потапово
@2
01
Ховрино
D3
@2
02
Беломорская
@2
03
Речной вокзал
@2
04
Водный стадион
@2
05
Войковская
14
14
D2
@2
06
Сокол
@2
07
Аэропорт
@2
08
Динамо
11
@2
09
Белорусская
05
D1
D4
@2
10
Маяковская
@2
11
Тверская
07
09
@2
12
Театральная
01
03
@2
13
Новокузнецкая
06
08
@2
14
Павелецкая
05
@2
15
Автозаводская
14
@2
16
Технопарк
@2
17
Коломенская
@2
18
11
@2
19
Кантемировская
@2
20
Царицыно
D2
@2
21
Орехово
@2
22
Домодедовская
@2
23

Широта    Долгота
0    55.814800  37.734200
1    55.803800  37.744800
2    55.796300  37.715100
3    55.788800  37.680200
4    55.780100  37.667300
..         ...        ...
264  55.626319  37.487647
265  55.616663  37.481290
266  55.616663  37.481290
267  55.616663  37.481290
268  55.616663  37.481290

[269 rows x 2 columns]

In [13]:
print(stations)

[<a class="mw-kartographer-maplink" data-lang="ru" data-lat="55.8148" data-lon="37.7342" data-mw-kartographer="maplink" data-overlays='["_4b326a8a0c01da3a7bf42178410fec4d10db58bb"]' data-style="osm-intl" data-zoom="14" href="/wiki/%D0%A1%D0%BB%D1%83%D0%B6%D0%B5%D0%B1%D0%BD%D0%B0%D1%8F:Map/14/55.8148/37.7342/ru">55°48′53″ с. ш. 37°44′03″ в. д.</a>, <a class="mw-kartographer-maplink" data-lang="ru" data-lat="55.8038" data-lon="37.7448" data-mw-kartographer="maplink" data-overlays='["_d1210d135010f191dd8fd92c4a206b1d82dcd23a"]' data-style="osm-intl" data-zoom="14" href="/wiki/%D0%A1%D0%BB%D1%83%D0%B6%D0%B5%D0%B1%D0%BD%D0%B0%D1%8F:Map/14/55.8038/37.7448/ru">55°48′14″ с. ш. 37°44′41″ в. д.</a>, <a class="mw-kartographer-maplink" data-lang="ru" data-lat="55.7963" data-lon="37.7151" data-mw-kartographer="maplink" data-overlays='["_25b112f51fbaff740c09a7641067406c26fb9d32"]' data-style="osm-intl" data-zoom="14" href="/wiki/%D0%A1%D0%BB%D1%83%D0%B6%D0%B5%D0%B1%D0%BD%D0%B0%D1%8F:Map/14/55.7963/3

In [15]:
filtered_stations = [item for item in names if item[0].isalpha() and item[0]!="D"]
print(filtered_stations)

['Бульвар Рокоссовского', 'Черкизовская', 'Преображенская площадь', 'Сокольники', 'Красносельская', 'Комсомольская', 'Красные Ворота', 'Чистые пруды', 'Лубянка', 'Охотный Ряд', 'Библиотека имени Ленина', 'Кропоткинская', 'Парк культуры', 'Фрунзенская', 'Спортивная', 'Воробьёвы горы', 'Университет', 'Проспект Вернадского', 'Юго-Западная', 'Тропарёво', 'Румянцево', 'Саларьево', 'Прокшино', 'Ольховая', 'Новомосковская', 'Потапово', 'Ховрино', 'Беломорская', 'Речной вокзал', 'Водный стадион', 'Войковская', 'Сокол', 'Аэропорт', 'Динамо', 'Белорусская', 'Маяковская', 'Тверская', 'Театральная', 'Новокузнецкая', 'Павелецкая', 'Автозаводская', 'Технопарк', 'Коломенская', 'Кантемировская', 'Царицыно', 'Орехово', 'Домодедовская', 'Красногвардейская', 'Алма-Атинская', 'Пятницкое шоссе', 'Митино', 'Волоколамская', 'Мякинино', 'Строгино', 'Крылатское', 'Молодёжная', 'Кунцевская', 'Славянский бульвар', 'Парк Победы', 'Киевская', 'Смоленская', 'Арбатская', 'Площадь Революции', 'Курская', 'Бауманская',

In [16]:
print(len(filtered_stations))

259


In [ ]:
from google.colab import files
files.upload()
df_metro = pd.read_csv('list_of_moscow_metro_stations.csv')


In [89]:
dfnext=dfs
display(dfnext)

Адрес Ближайшее метро  \
6     Москва, САО, р-н Хорошевский, 1-я Магистральна...    Полежаевская   
7     Москва, СЗАО, р-н Хорошево-Мневники, 3-я Хорош...           Зорге   
8     Москва, СЗАО, р-н Хорошево-Мневники, 3-я Хорош...           Зорге   
9     Москва, СЗАО, р-н Хорошево-Мневники, 3-я Хорош...           Зорге   
10    Москва, ЗАО, р-н Можайский, Верейская 41 жилой...       Давыдково   
...                                                 ...             ...   
1466  Москва, ЗАО, р-н Фили-Давыдково, Большая Филев...      Пионерская   
1469    Москва, САО, р-н Савеловский, Вятская ул., 27С7     Савёловская   
1470  Москва, ЦАО, р-н Замоскворечье, Садовническая ...   Новокузнецкая   
1471           Москва, ЦАО, р-н Якиманка, Житная ул., 4    Добрынинская   
1474  Москва, СЗАО, р-н Покровское-Стрешнево, Волоко...         Спартак   

                  Цена   Площадь                 Район Округ  Чистая цена  \
6       299 389 ₽/мес.   80,7 м²           Хорошевский   САО       299389   
7       544 335 ₽/мес.  153,1 м²     Хорошево-Мневники  СЗАО       544335   
8     1 980 718 ₽/мес.  696,7 м²     Хорошево-Мневники  СЗАО      1980718   
9       681 462 ₽/мес.  196,2 м²     Хорошево-Мневники  СЗАО       681462   
10      388 316 ₽/мес.  104,5 м²             Можайский   ЗАО       388316   
...                ...       ...                   ...   ...          ...   
1466    400 000 ₽/мес.    250 м²        Фили-Давыдково   ЗАО       400000   
1469  1 257 600 ₽/мес.    262 м²           Савеловский   САО      1257600   
1470  2 500 000 ₽/мес.    750 м²         Замоскворечье   ЦАО      2500000   
1471  4 100 000 ₽/мес.  585,5 м²              Якиманка   ЦАО      4100000   
1474    250 000 ₽/мес.     50 м²  Покровское-Стрешнево  СЗАО       250000   

      Чистая площадь    Цена за м  
6               80.7  3709.900867  
7              153.1  3555.421293  
8              696.7  2842.999856  
9              196.2  3473.302752  
10             104.5  3715.942584  
...              ...          ...  
1466           250.0  1600.000000  
1469           262.0  4800.000000  
1470           750.0  3333.333333  
1471           585.5  7002.561913  
1474            50.0  5000.000000  

[1387 rows x 9 columns]

In [140]:
dfnext = dfnext[dfnext['Округ']=='ЦАО']
dfnext = dfnext[dfnext['Чистая площадь']<120]
dfnext = dfnext[dfnext['Чистая площадь']>75]
dfnext = dfnext[dfnext["Ближайшее метро"].isin(["Павелецкая", "Улица 1905 года", "Белорусская", "Смоленская", "Добрынинская", "Шаболовская", "Тульская"])]
dfnext = dfnext[dfnext['Цена за м']>2000]
dfnext = dfnext[dfnext['Цена за м']<8000]
dfnext=dfnext.drop(index=898)

In [141]:
display(dfnext)

Адрес  Ближайшее метро  \
86    Москва, ЦАО, р-н Тверской, 1-я Тверская-Ямская...      Белорусская   
160   Москва, ЦАО, р-н Пресненский, Звенигородское ш...  Улица 1905 года   
221   Москва, ЦАО, р-н Замоскворечье, Люсиновская ул...     Добрынинская   
656         Москва, ЦАО, р-н Арбат, ул. Новый Арбат, 19       Смоленская   
830    Москва, ЦАО, р-н Тверской, ул. Бутырский Вал, 68      Белорусская   
895           Москва, ЦАО, р-н Арбат, ул. Арбат, 30/3С1       Смоленская   
917           Москва, ЦАО, р-н Арбат, ул. Арбат, 30/3С1       Смоленская   
1024  Москва, ЦАО, р-н Замоскворечье, Валовая ул., 2...       Павелецкая   
1055  Москва, ЦАО, р-н Пресненский, Большая Грузинск...      Белорусская   
1063  Москва, ЦАО, р-н Пресненский, Звенигородское ш...  Улица 1905 года   
1228  Москва, ЦАО, р-н Тверской, 1-я Тверская-Ямская...      Белорусская   

                Цена   Площадь          Район Округ  Чистая цена  \
86    849 000 ₽/мес.    112 м²       Тверской   ЦАО       849000   
160   500 000 ₽/мес.  112,1 м²    Пресненский   ЦАО       500000   
221   320 000 ₽/мес.   85,7 м²  Замоскворечье   ЦАО       320000   
656   242 317 ₽/мес.   93,8 м²          Арбат   ЦАО       242317   
830   350 000 ₽/мес.    100 м²       Тверской   ЦАО       350000   
895   395 000 ₽/мес.     90 м²          Арбат   ЦАО       395000   
917   400 000 ₽/мес.     90 м²          Арбат   ЦАО       400000   
1024  475 000 ₽/мес.    111 м²  Замоскворечье   ЦАО       475000   
1055  500 000 ₽/мес.     86 м²    Пресненский   ЦАО       500000   
1063  500 004 ₽/мес.  112,1 м²    Пресненский   ЦАО       500004   
1228  650 000 ₽/мес.    112 м²       Тверской   ЦАО       650000   

      Чистая площадь    Цена за м  
86             112.0  7580.357143  
160            112.1  4460.303301  
221             85.7  3733.955659  
656             93.8  2583.336887  
830            100.0  3500.000000  
895             90.0  4388.888889  
917             90.0  4444.444444  
1024           111.0  4279.279279  
1055            86.0  5813.953488  
1063           112.1  4460.338983  
1228           112.0  5803.571429

In [101]:
print(*dfnext["Адрес"].to_list(), sep="\n")

Москва, ЦАО, р-н Тверской, 1-я Тверская-Ямская ул., 25С1
Москва, ЦАО, р-н Пресненский, Звенигородское ш., 18/20к1
Москва, ЦАО, р-н Замоскворечье, Люсиновская ул., 36С1
Москва, ЦАО, р-н Арбат, ул. Новый Арбат, 19
Москва, ЦАО, р-н Тверской, ул. Бутырский Вал, 68
Москва, ЦАО, р-н Арбат, ул. Арбат, 30/3С1
Москва, ЦАО, р-н Арбат, ул. Арбат, 30/3С1
Москва, ЦАО, р-н Арбат, ул. Арбат, 30/3С1
Москва, ЦАО, р-н Замоскворечье, Валовая ул., 21К125
Москва, ЦАО, р-н Пресненский, Большая Грузинская ул., 56
Москва, ЦАО, р-н Пресненский, Звенигородское ш., 18/20к1
Москва, ЦАО, р-н Тверской, 1-я Тверская-Ямская ул., 27


<a class="mw-kartographer-maplink" data-lang="ru" data-lat="55.7801" data-lon="37.6673"